ruby-challenges.rb

```diff
@@ -10,20 +10,93 @@
 
+# create a method named letter_contain that takes an array and a string as parameters
+def letter_contain(array, string)
+  # use .select to iterate over the array and remove values that do not meet certain conditions
+  # use variable word for the string being iterated over
+  # use .include? to see if the word has the string and print 
+  puts array.select {|word| word.include?(string)}
+end
+
+letter_contain(beverages_array, letter_o)
+# Output: 
+# coffee
+# soda water
+
+letter_contain(beverages_array, letter_t)
+# Output: 
+# water
+# soda water
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+def hash_value(hash)
+  puts hash.values.sort
+end
+
+hash_value(us_states)
+# Output: 
+# California
+# Arizona
+# Nevada
+# Maine
+# New Hampshire
+# Rhode Island
+# Washington
+# Oregon
+# Idaho
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+# create a class called bike
+class Bike 
+  # Bike is initialized with the bikes model name
+  def initialize(model)
+    # by default, the bike starts at 0 speed and has 2 wheels. @model recieves the name from model
+    @wheels = 2
+    @current_speed = 0
+    @model = model
+  end
+  # bike_info is a class method that recieves the initialized variables and prints 
+  # a string when invoked using string interpolation
+  def bike_info
+    p"The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph"
+  end
+  # pedal_faster is a class method that recieves a number in the 'speed' parameter
+  def pedal_faster(speed)
+    # we only want positive numbers for increased speed so it has to be greater 
+    # than zero
+    if(speed>0)
+      # we add the speed number on to our current_speed
+     p @current_speed += speed
+    end
+  end
+  # brake is a class method that also recieves the speed parameter
+  # in this case it will be the speed deducted from our current_speed
+  def brake(speed)
+    # if the current_speed - speed would be less than zero - then just make
+    # current speed 0, also we want only positive numbers
+    if(@current_speed - speed < 0 && speed > 0)
+     p @current_speed = 0
+    else
+     p @current_speed -= speed
```

Comment:
I see you're trying to handle edge cases, which is great! Also, I noticed you are printing each return in your methods. This is fine if you want to see the result every time you call the method, which can be seen below on line 95, for example. If you didn't want to print the output every time you call `pedal_faster` and `brake`, you can remove the `p` in those methods and just use `p` on the method call. Just something to keep in mind for later on when you are building your full stack apps!